src/main.ml

```diff
@@ -1223,7 +1242,44 @@ let old_main () =
       hip_epilogue ()
     end
 
+let test_api () =
+  let () = print_string "\n TESTING API" in
+
+  (* true |- true *)
+  let true_f = Sleekapi.bool_pure_f true in
+  let empty_heap_f = Sleekapi.empty_heap_f () in
+  let ante_f = Sleekapi.ante_f empty_heap_f true_f in
+  let conseq_f = Sleekapi.conseq_f empty_heap_f true_f in
+  let () = print_string "\n Entail 1: \n" in
+  let () = print_string (Sleekapi.ante_printer ante_f) in
+  let () = print_string (Sleekapi.conseq_printer conseq_f) in
+  let () = print_string ("\n ENTAIL RESULT : " ^ (string_of_bool (Sleekapi.entail ante_f conseq_f))) in
+
+  (* x > 0 /\ y = x + 1 |- y > 1 *)
+  let ante_f = Sleekapi.ante_f empty_heap_f
+      (Sleekapi.and_f
+         (Sleekapi.gt_pure_f 
+            (Sleekapi.var_pure_exp "x" false)
+            (Sleekapi.int_pure_exp 0))
+         (Sleekapi.eq_pure_f
+            (Sleekapi.var_pure_exp "y" false)
+            (Sleekapi.add_pure_exp
+               (Sleekapi.var_pure_exp "x" false)
+               (Sleekapi.int_pure_exp 1)))) in
+  let conseq_f = Sleekapi.conseq_f empty_heap_f
+      (Sleekapi.gt_pure_f
+         (Sleekapi.var_pure_exp "y" false)
+         (Sleekapi.int_pure_exp 1)) in
+  let () = print_string "\n Entail 2: \n" in
+  let () = print_string (Sleekapi.ante_printer ante_f) in
+  let () = print_string (Sleekapi.conseq_printer conseq_f) in
+  let () = print_string ("\n ENTAIL RESULT : " ^ (string_of_bool (Sleekapi.entail ante_f conseq_f))) in
+            
+  let () = print_string "\n API TEST COMPLETE" in
```

Comment:
Now that we are using dune, we can probably start setting up some [tests](https://dune.readthedocs.io/en/stable/tests.html#inline-expectation-tests).